### PR TITLE
SRV resolving / Small Handshake rework

### PR DESCRIPTION
**Pull request by ChickenPat**
- Add SRV resolving for standalone / legacy ping passthrough
- Handshake now uses the server address directly from the config and no longer the IP from a domain (Some servers use the address that is given during the handshake)

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 1]
```diff
@@ -136,8 +141,28 @@ private GeyserConnector(PlatformType platformType, GeyserBootstrap bootstrap) {
...
+                Attribute attr = ctx.getAttributes("dns:///_minecraft._tcp." + remoteAddress, new String[]{"SRV"}).get("SRV");
+                if (attr.size() > 0) {
+                    String[] record = ((String) attr.get(0)).split(" ");
+                    remoteAddress = record[3];
+                    if(remoteAddress.endsWith(".")){
+                        remoteAddress = remoteAddress.substring(0, remoteAddress.length() - 1);
+                    }
+                    config.getRemote().setAddress(remoteAddress);
+                    config.getRemote().setPort(remotePort = Integer.parseInt(record[2]));
+                    logger.debug(LanguageUtils.getLocaleStringLog("geyser.core.start.srv_record", remoteAddress, String.valueOf(remotePort)));
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:158]: Debug strings don't need to be translated.

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 2]
```diff
@@ -136,8 +141,28 @@ private GeyserConnector(PlatformType platformType, GeyserBootstrap bootstrap) {
         if (platformType != PlatformType.STANDALONE) {
             DockerCheck.check(bootstrap);
         }
+        String remoteAddress = config.getRemote().getAddress();
+        int remotePort = config.getRemote().getPort();
+        if((config.isLegacyPingPassthrough() || platformType == PlatformType.STANDALONE) && !remoteAddress.matches(IP_REGEX) && !remoteAddress.equalsIgnoreCase("localhost")){
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:146]: Use spacing `if (condition) {` to match with the rest of the Geyser codebase.

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 3]
```diff
@@ -136,8 +141,28 @@ private GeyserConnector(PlatformType platformType, GeyserBootstrap bootstrap) {
...
+                    String[] record = ((String) attr.get(0)).split(" ");
+                    remoteAddress = record[3];
+                    if(remoteAddress.endsWith(".")){
+                        remoteAddress = remoteAddress.substring(0, remoteAddress.length() - 1);
+                    }
+                    config.getRemote().setAddress(remoteAddress);
+                    config.getRemote().setPort(remotePort = Integer.parseInt(record[2]));
+                    logger.debug(LanguageUtils.getLocaleStringLog("geyser.core.start.srv_record", remoteAddress, String.valueOf(remotePort)));
+                }
+            }catch (NamingException ex){
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:160]: Use spacing (`} catch (NamingException ex) {`)to match with the rest of the Geyser codebase.

**Comment by ChickenPat** [comment]
I implemented all suggestions

**Comment by rtm516** [comment]
Not entirely sure why `InitialDirContext` was used instead of `DnsNameResolver` from Netty but it all works fine, and should be good for a merge.

**Comment by DoctorMacc** [comment]
Thank you for your contribution!

**Pull request merged**